docs/modules/maintain-cluster/pages/enterprise-rest-api.adoc

```diff
@@ -474,3 +474,104 @@ curl -X 'DELETE' \
   "message": "CP Subsystem is not enabled!"
 }
 ----
+
+== Dynamic Configuration Update REST Endpoint
+You can use the `/hazelcast/rest/api/v1/config/update` REST endpoint to change the dynamic server configurations (see xref:configuration:dynamic-config.adoc[Dynamic Configuration for Members]). The dynamic configuration lets you change different configurations in addition to adding new configurations for Hazelcast data structures.
+
+The endpoint requires that you send a xml/yaml server configuration file with the needed changes. The response will be two lists in json format. The first list is the newly added configurations among the configs sent to the server, and another list of ignored configurations which were in the sent configuration but could not be applied at the server, possibly since they are not dynamic configuration options but static configurations.
```

Comment:
I think we need to update the info about the ignored configurations.

After running the endpoint multiple times, I observed the following behaviors:
1. If I add the same dynamic config more than once but in different requests, duplicate config is placed in ignored list.
2. If I try to add duplicate dynamic configs within the single request, `400` response is returned with a message `"interpretation error: the resource is neither valid XML nor valid YAML"`
3. If I try to add static configs, they are not placed in ignored list but also they are not placed in added list as well. Response is like below:
```
{
  "addedConfigs": [],
  "ignoredConfigs": []
}
```

Reply:
Answers:

1. We have this already in the config. I made different sections to highlight this now.
2. I tried this duplicate map config and received 200 OK:
```
curl -X 'POST' \
  'http://localhost:49567/hazelcast/rest/api/v1/cluster/config/update' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJyb2xlcyI6IltcImFkbWluXCJdIiwiaWF0IjoxNzE3NTg5ODE3LCJleHAiOjE3MTc1OTA3MTd9.2VVyxXPiF5710qw0VDwUGEHYh52nSdrsuQfcwyOkBeM' \
  -H 'Content-Type: application/json' \
  -d '<hazelcast xmlns="http://www.hazelcast.com/schema/config">
                       <map name="duplicate-map">
                           <statistics-enabled>true</statistics-enabled>
                       </map>
                       <map name="duplicate-map">
                           <statistics-enabled>true</statistics-enabled>
                       </map>
        </hazelcast>'
```
And I got this response:
```
{
  "addedConfigs": [
    {
      "sectionName": "map",
      "configName": "duplicate-map"
    }
  ],
  "ignoredConfigs": []
}
```

Can you share the example in your case? May be your XML is not correct.

3. Added this example.

Please see 7b5f64f375d28d2a34bd0ea0a0026ca39f43a406 for changes as suggested.

Reply:
You are correct. I trued this yaml file to start the server 
```yaml
hazelcast:
  map:
    my-map-yaml:
      in-memory-format: BINARY
      statistics-enabled: true
      backup-count: 2
    my-map-yaml:
      in-memory-format: BINARY
      statistics-enabled: true
      backup-count: 2
```
And the server fails with this message:
```
INFO: Loading configuration '/Users/ihsan/Desktop/hazelcast.yml' from System property 'hazelcast.config'
Jun 06, 2024 11:41:58 AM com.hazelcast.internal.config.AbstractConfigLocator
INFO: Using configuration file at /Users/ihsan/Desktop/hazelcast.yml
Exception in thread "main" com.hazelcast.config.InvalidConfigurationException: Invalid YAML configuration
	at com.hazelcast.config.YamlConfigBuilder.parseAndBuildConfig(YamlConfigBuilder.java:152)
	at com.hazelcast.config.YamlConfigBuilder.build(YamlConfigBuilder.java:138)
	at com.hazelcast.config.YamlConfigBuilder.build(YamlConfigBuilder.java:131)
	at com.hazelcast.config.Config.loadFromFile(Config.java:297)
	at com.hazelcast.config.Config.load(Config.java:261)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:132)
	at com.hazelcast.core.Hazelcast.newHazelcastInstance(Hazelcast.java:95)
	at com.hazelcast.core.server.HazelcastMemberStarter.main(HazelcastMemberStarter.java:45)
Caused by: com.hazelcast.internal.yaml.YamlException: An error occurred while loading and parsing the YAML stream
	at com.hazelcast.internal.yaml.YamlLoader.load(YamlLoader.java:71)
	at com.hazelcast.config.YamlConfigBuilder.parseAndBuildConfig(YamlConfigBuilder.java:150)
	... 7 more
Caused by: while constructing a mapping
 in reader, line 3, column 5:
        my-map-yaml:
        ^
found duplicate key my-map-yaml
 in reader, line 7, column 5:
        my-map-yaml:
        ^

	at org.snakeyaml.engine.v2.constructor.StandardConstructor.processDuplicateKeys(StandardConstructor.java:90)
	at org.snakeyaml.engine.v2.constructor.StandardConstructor.flattenMapping(StandardConstructor.java:70)
	at org.snakeyaml.engine.v2.constructor.StandardConstructor.constructMapping2ndStep(StandardConstructor.java:119)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructMapping(BaseConstructor.java:278)
	at org.snakeyaml.engine.v2.constructor.StandardConstructor$ConstructYamlMap.construct(StandardConstructor.java:203)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructObjectNoCheck(BaseConstructor.java:153)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructObject(BaseConstructor.java:133)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructMapping2ndStep(BaseConstructor.java:302)
	at org.snakeyaml.engine.v2.constructor.StandardConstructor.constructMapping2ndStep(StandardConstructor.java:120)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructMapping(BaseConstructor.java:278)
	at org.snakeyaml.engine.v2.constructor.StandardConstructor$ConstructYamlMap.construct(StandardConstructor.java:203)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructObjectNoCheck(BaseConstructor.java:153)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructObject(BaseConstructor.java:133)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructMapping2ndStep(BaseConstructor.java:302)
	at org.snakeyaml.engine.v2.constructor.StandardConstructor.constructMapping2ndStep(StandardConstructor.java:120)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructMapping(BaseConstructor.java:278)
	at org.snakeyaml.engine.v2.constructor.StandardConstructor$ConstructYamlMap.construct(StandardConstructor.java:203)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructObjectNoCheck(BaseConstructor.java:153)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructObject(BaseConstructor.java:133)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.construct(BaseConstructor.java:92)
	at org.snakeyaml.engine.v2.constructor.BaseConstructor.constructSingleDocument(BaseConstructor.java:79)
	at org.snakeyaml.engine.v2.api.Load.loadOne(Load.java:111)
	at org.snakeyaml.engine.v2.api.Load.loadFromInputStream(Load.java:123)
	at com.hazelcast.internal.yaml.YamlLoader.load(YamlLoader.java:68)
	... 8 more
```

We did not set the [allow duplicate keys option](https://javadoc.io/static/org.snakeyaml/snakeyaml-engine/2.7/org/snakeyaml/engine/v2/api/LoadSettingsBuilder.html#setAllowDuplicateKeys(boolean)) in the yaml engine `LoadSettingsBuilder` hence it is false and fails by default. If I set it to true the server starts similar to the XML duplicate config. Hence, there is a behaviour difference at the server side for duplicate keys.